packages/aws-cdk-lib/aws-cloudfront/lib/realtime-log-config.ts

```diff
@@ -0,0 +1,144 @@
+ */
+function singletonKinesisRole(scope: Construct): iam.IRole {
+  const id = 'RealtimeLogKinesisRole';
+  const existing = scope.node.tryFindChild(id) as iam.IRole;
+  if (existing) { return existing; }
+
+  const role = new iam.Role(scope, id, {
+    roleName: PhysicalName.GENERATE_IF_NEEDED,
+    assumedBy: new iam.ServicePrincipal('cloudfront.amazonaws.com'),
+  });
+
+  return role;
+}
+
+/**
+ * Represents the endpoints available for targetting within a realtime log config resource
+ */
+export abstract class Endpoint {
+  /**
+   * Configure a Kinesis Stream Endpoint for Realtime Log Config
+   *
+   * @default - a role will be created and used across your endpoints
+   */
+  public static fromKinesisStream(stream: kinesis.IStream, role?: iam.IRole): Endpoint {
+    return new (class extends Endpoint {
+      public _renderEndpoint(scope: Construct) {
+        const cloudfrontRole = role ?? singletonKinesisRole(scope);
+
+        stream.grant(cloudfrontRole,
+          'kinesis:DescribeStreamSummary',
+          'kinesis:DescribeStream',
+          'kinesis:PutRecord',
+          'kinesis:PutRecords',
+        );
+
+        if (stream.encryptionKey) {
+          stream.encryptionKey.grant(cloudfrontRole, 'kms:GenerateDataKey');
+        }
+
+        return {
+          kinesisStreamConfig: {
+            roleArn: cloudfrontRole.roleArn,
+            streamArn: stream.streamArn,
+          },
+          streamType: 'Kinesis',
+        };
+      }
+    });
+  }
+
+  private constructor() {}
+
+  /**
+  * @internal
+  */
+  public abstract _renderEndpoint(scope: Construct): any;
+}
+
+/**
+ * Represents Realtime Log Configuration
+ */
+export interface IRealtimeLogConfig extends IResource {
+  /**
+   * The name of the realtime log config.
+   * @attribute
+   */
+  readonly realtimeLogConfigName: string;
+  /**
+   * The arn of the realtime log config.
+   * @attribute
+   */
+  readonly realtimeLogConfigArn: string;
+}
+
+/**
+ * Properties for defining a RealtimeLogConfig resource.
+ */
+export interface RealtimeLogConfigProps {
+  /**
+   * The unique name of this real-time log configuration.
+   *
+   * @default - the unique construct ID
+   */
+  readonly realtimeLogConfigName?: string;
+  /**
+   * A list of fields that are included in each real-time log record.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/real-time-logs.html#understand-real-time-log-config-fields
+   */
+  readonly fields: string[];
+  /**
+   * The sampling rate for this real-time log configuration.
+   */
+  readonly samplingRate: number;
+  /**
+   * Contains information about the Amazon Kinesis data stream where you are sending real-time log data for this real-time log configuration.
+   */
+  readonly endPoints: Endpoint[];
+}
+
+/**
+ * A Realtime Log Config configuration
+ *
+ * @resource AWS::CloudFront::RealtimeLogConfig
+ */
+export class RealtimeLogConfig extends Resource implements IRealtimeLogConfig {
```

Comment:
I think this class needs the static methods that allow a user to import an existing `RealtimeLogConfig`. We generally have 3 - `fromRealtimeLogConfigName`, `fromRealtimeLogConfigArn`, `fromRealtimeLogConfigAttributes` - for example - https://github.com/aws/aws-cdk/blob/27b720135e30d58732997a5a5e172daaa9453381/packages/aws-cdk-lib/aws-iam/lib/instance-profile.ts#L113-L157

Reply:
Added them